In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

In [2]:
import numpy as np
import pandas as pd

In [31]:
df = pd.read_csv('/kaggle/input/titanic/train.csv')

In [32]:
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [33]:
x = df.drop(['PassengerId', 'Name', 'Ticket', 'Cabin', 'Embarked', 'Survived', 'Age', 'Fare'], axis='columns')

In [34]:
x.head()

Pclass     Sex  SibSp  Parch
0       3    male      1      0
1       1  female      1      0
2       3  female      0      0
3       1  female      1      0
4       3    male      0      0

In [35]:
y = df.Survived

In [36]:
y.head()

0    0
1    1
2    1
3    1
4    0
Name: Survived, dtype: int64

In [37]:
x.head()

Pclass     Sex  SibSp  Parch
0       3    male      1      0
1       1  female      1      0
2       3  female      0      0
3       1  female      1      0
4       3    male      0      0

In [38]:
from sklearn import preprocessing
from sklearn.preprocessing import LabelEncoder


In [39]:
label_encoder = preprocessing.LabelEncoder()
x['Sex'] = label_encoder.fit_transform(x['Sex'])
x

Pclass  Sex  SibSp  Parch
0         3    1      1      0
1         1    0      1      0
2         3    0      0      0
3         1    0      1      0
4         3    1      0      0
..      ...  ...    ...    ...
886       2    1      0      0
887       1    0      0      0
888       3    0      1      2
889       1    1      0      0
890       3    1      0      0

[891 rows x 4 columns]

In [40]:
x.isnull().sum()

Pclass    0
Sex       0
SibSp     0
Parch     0
dtype: int64

In [41]:
x.isnull().sum()

Pclass    0
Sex       0
SibSp     0
Parch     0
dtype: int64

In [42]:
from sklearn.model_selection import train_test_split

In [43]:
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.2, random_state=1)

In [44]:
from xgboost import XGBClassifier

In [45]:
model = XGBClassifier(n_estimators = 1000, learning_rate = 0.001, max_depth = 15, random_state=1)

In [46]:
model.fit(x, y)

XGBClassifier(base_score=None, booster=None, callbacks=None,
              colsample_bylevel=None, colsample_bynode=None,
              colsample_bytree=None, device=None, early_stopping_rounds=None,
              enable_categorical=False, eval_metric=None, feature_types=None,
              gamma=None, grow_policy=None, importance_type=None,
              interaction_constraints=None, learning_rate=0.001, max_bin=None,
              max_cat_threshold=None, max_cat_to_onehot=None,
              max_delta_step=None, max_depth=15, max_leaves=None,
              min_child_weight=None, missing=nan, monotone_constraints=None,
              multi_strategy=None, n_estimators=1000, n_jobs=None,
              num_parallel_tree=None, random_state=1, ...)

In [47]:
testdata = pd.read_csv('/kaggle/input/titanic/test.csv')

In [48]:
testdata.head()

PassengerId  Pclass                                          Name     Sex  \
0          892       3                              Kelly, Mr. James    male   
1          893       3              Wilkes, Mrs. James (Ellen Needs)  female   
2          894       2                     Myles, Mr. Thomas Francis    male   
3          895       3                              Wirz, Mr. Albert    male   
4          896       3  Hirvonen, Mrs. Alexander (Helga E Lindqvist)  female   

    Age  SibSp  Parch   Ticket     Fare Cabin Embarked  
0  34.5      0      0   330911   7.8292   NaN        Q  
1  47.0      1      0   363272   7.0000   NaN        S  
2  62.0      0      0   240276   9.6875   NaN        Q  
3  27.0      0      0   315154   8.6625   NaN        S  
4  22.0      1      1  3101298  12.2875   NaN        S

In [49]:
test = testdata.drop(['PassengerId', 'Name', 'Ticket', 'Cabin', 'Embarked', 'Age', 'Fare'], axis='columns')

In [50]:
test.head()

Pclass     Sex  SibSp  Parch
0       3    male      0      0
1       3  female      1      0
2       2    male      0      0
3       3    male      0      0
4       3  female      1      1

In [51]:
label_encoder1 = preprocessing.LabelEncoder()
test['Sex'] = label_encoder1.fit_transform(test['Sex'])
test.head()

Pclass  Sex  SibSp  Parch
0       3    1      0      0
1       3    0      1      0
2       2    1      0      0
3       3    1      0      0
4       3    0      1      1

In [52]:
test.isnull().sum()

Pclass    0
Sex       0
SibSp     0
Parch     0
dtype: int64

In [54]:
test.isnull().sum()

Pclass    0
Sex       0
SibSp     0
Parch     0
dtype: int64

In [55]:
predictions = model.predict(test)

In [56]:
output = pd.DataFrame({'PassengerId': testdata['PassengerId'], 'Survived':predictions})
output.to_csv('submission.csv', index=False)

# Review

Type of problem - Classification Problem
<br>
Type of dataset - supervised learning dataset bcz i have both feature and targeted value
<br>
Number of Instances - 891 X 12
<br>
For this classification task i drop following features ['PassengerId', 'Name', 'Ticket', 'Cabin', 'Embarked', 'Survived', 'Age', 'Fare'] because for prediction of target this attributes are not make such impact so after trail and error this attributes are dropped to make the training dataset.
<br>
To feed my data into the classification model first i do some preprocessing like there are categorical values as well that are not numeric so i use LabelEncoder there because i do the encoading of attribute ['sex'] as it have only two possibilities male and female thus i use the label encoader there.
<br>
After preprocess the data i feed the data into the XGBClassifier Model and generate the submission file.